### DISCUSSION TITLE
Comparing eGRNs between genotypes

### DISCUSSION THREAD

**Discussion by Raghav1881** [aertslab/scenicplus#278] [Q&A]
Hi,

I have a multiome object derived from the mouse hippocampus and I am processing the object according to the melanoma cell lines vignette. Within my dataset metadata, I have a category for cluster annotations named 'region' which corresponds to anatomical regions and cell types within the hippocampus (e.g. CA1, dentate gyrus, inhibitory, etc.), and another category for my treatments (i.e. wildtype, heterozygous-knockout, homozygous-knockout). 

I was wondering if I can generate eGRNs across treatments (i.e. wildtype vs heterozygous-knockout, wildtype vs homozygous-knockout) within each of the regions (regions being clusters for hippocampal regions). So for example, comparing treatments within the CA1 region and having eGRNs corresponding to the comparisons. I was considering taking a subsetting approach, by subsetting all of the objects by regions, then doing the comparisons, but wasn't sure if there was an easier way to go about it.

To further add, when calculating DARs, how should my contrasts be set up to compare the treatments mentioned above (WT vs HET, WT vs KO).

Thank you

**Comment by SeppeDeWinter** [comment]
Hi @Raghav1881 

This is not a straightforward task. 
I would recommend to run SCENIC+ one time on the entire dataset and do the comparisons afterwards. There is, however, not one good way to do this it depends on the particular biological question you are interested in.

As for the DARS. The easiest is to create a new variable merging the labels of the region and the condition, (for example CA1_WT, CA1_HET, CA1_KO). After this you can setup a contrast for each condition and region like this:

```python

contrasts = [
      [ ["CA1_WT"], ["CA1_KO"]   ], #contrast 1
      [ ["CA1_WT"], ["CA1_HET"] ], # contrast 2
      ...
]

```

I hope this helps.

Good luck with your analysis.

All the best,

Seppe

**Reply by Raghav1881** [reply]
Hi Seppe,

Appreciate the prompt response, thanks for the clarification. Just to make sure, when you say to run SCENIC+ on the entire dataset, at the pycistopic/pycistarget object step, should my region sets only contain topics and DARs across all regions in my dataset (i.e. run find_diff_features using the variable 'region')?

Thanks again, and I greatly appreciate your work in SCENIC+ and related packages,

Raghav

**Reply by SeppeDeWinter** [reply]
Hi Raghav

What I meant was running SCENIC+ on all the _cells_, i.e.  don't do seperate runs on different subset of cells. For the region sets, I would try to include as many comparisons as possible (so **not** only for the variable `region`).

I hope this clarifies things.

All the best,

Seppe
